embeddings/task-type-embedding.ipynb

```diff
@@ -0,0 +1,3079 @@
+       "      quickchartButtonEl.classList.add('colab-df-spinner');\n",
+       "      try {\n",
+       "        const charts = await google.colab.kernel.invokeFunction(\n",
+       "            'suggestCharts', [key], {});\n",
+       "      } catch (error) {\n",
+       "        console.error('Error during call to suggestCharts:', error);\n",
+       "      }\n",
+       "      quickchartButtonEl.classList.remove('colab-df-spinner');\n",
+       "      quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
+       "    }\n",
+       "    (() => {\n",
+       "      let quickchartButtonEl =\n",
+       "        document.querySelector('#df-68ec3e8c-b73c-48e8-8938-f885d8e7d561 button');\n",
+       "      quickchartButtonEl.style.display =\n",
+       "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
+       "    })();\n",
+       "  </script>\n",
+       "</div>\n",
+       "    </div>\n",
+       "  </div>\n"
+      ]
+     },
+     "metadata": {},
+     "execution_count": 7
+    }
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "### Calculate cosine similarities between questions and answers\n",
+    "\n",
+    "Next, we will calculate the cosine similarity for all combinations of the questions and answers using `cosine_similarity()` function from scikit-learn. Thus, the `calc_cosine_similarities()` function will return a two dimensional array with similarities between 1K questions x 1K answers. By printing the result for the first question, you will see 1K values that represents the cosine similarity between the first question and the 1K answers."
+   ],
+   "metadata": {
+    "id": "qFzeZmpgzVDJ"
+   },
+   "id": "qFzeZmpgzVDJ"
+  },
+  {
+   "cell_type": "code",
+   "source": [
+    "from sklearn.metrics.pairwise import cosine_similarity\n",
+    "\n",
+    "\n",
+    "# calculate similarities for all combinations of the questions and answers\n",
+    "def calc_cosine_similarities(df):\n",
+    "    q_embs = df[\"q_emb\"].tolist()\n",
+    "    a_embs = df[\"a_emb\"].tolist()\n",
+    "    return cosine_similarity(q_embs, a_embs)\n",
+    "\n",
+    "\n",
+    "# print the answer similarities for the first question\n",
+    "q_and_a_similarities = calc_cosine_similarities(similarity_df)\n",
+    "q_and_a_similarities[0]"
+   ],
+   "metadata": {
+    "id": "5sin1iKsbGsm",
+    "colab": {
+     "base_uri": "https://localhost:8080/"
+    },
+    "executionInfo": {
+     "status": "ok",
+     "timestamp": 1726043581086,
+     "user_tz": -540,
+     "elapsed": 2137,
+     "user": {
+      "displayName": "",
+      "userId": ""
+     }
+    },
+    "outputId": "8486c694-45a0-4322-9838-14bcd229feb3"
+   },
+   "id": "5sin1iKsbGsm",
+   "execution_count": 8,
+   "outputs": [
+    {
+     "output_type": "execute_result",
+     "data": {
+      "text/plain": [
+       "array([0.39343315, 0.37640625, 0.33152983, 0.30465271, 0.36603535,\n",
```

Comment:
Can you change this to print out a smaller section of the embeddings array?

Reply:
fixed!